src/meta/src/rpc/ddl_controller.rs

```diff
@@ -609,6 +616,38 @@ impl DdlController {
         }
     }
 
+    async fn create_secret(&self, mut secret: Secret) -> MetaResult<NotificationVersion> {
+        // The 'secret' part of the request we receive from the frontend is in plaintext;
+        // here, we need to encrypt it before storing it in the catalog.
+
+        let encrypted_payload = simplestcrypt::encrypt_and_serialize(
+            self.env.opts.secret_store_private_key.as_slice(),
+            secret.get_value().as_slice(),
+        )
```

Comment:
we do the encryption and the secret key here. 
@yuhao-su

Reply:
Why not using the implementation in our built-in function `encrypt(bytea, bytea, varchar) -> bytea`?

https://github.com/risingwavelabs/risingwave/blob/1cc36e1a6a59f039a6031c78f3f8096f9e49f17b/src/expr/impl/src/scalar/encrypt.rs#L141-L157

Benefits include

- Avoid duplicates in dependencies
- Easy to encrypt & decrypt with SQL interface when debugging

Reply:
Agree with https://github.com/risingwavelabs/risingwave/pull/16288#discussion_r1608993469

> Better to consider a more popular encryption lib

It's a bit risky to use a 3rd-party unpopular lib for decryption & encryption here. Considering the possibility that the lib author may discontinue its maintenance, we should either use ourselves's implementation, or ensure we know 100% how the lib works so that we can write an exactly same one. The former one is easier, I think.

Reply:
> Why not using the implementation in our built-in function encrypt(bytea, bytea, varchar) -> bytea?

Good point! Let me fix it.

Reply:
https://github.com/risingwavelabs/risingwave/blob/af892fa4f4b5f7f2cddc1e2155132aa3bac32ffd/src/meta/src/rpc/ddl_controller.rs#L633-L660 
Impl the encryption myself :-)